state/migrations/20231108122539_clear_stuck_invites.go

```diff
@@ -0,0 +1,83 @@
+package migrations
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/lib/pq"
+	"github.com/pressly/goose/v3"
+)
+
+func init() {
+	goose.AddMigrationContext(upClearStuckInvites, downClearStuckInvites)
+}
+
+// The purpose of this migration is to find users who have rooms which have
+// not been properly processed by the proxy and invalidate their since token
+// so they will do an initial sync on the next poller startup. This is specifically
+// targeting stuck invites, where there is an invite in the invites table but
+// the room is already joined. This is usually (always?) due to missing a create
+// event when the room was joined, caused by a synapse bug outlined in
+// https://github.com/matrix-org/sliding-sync/issues/367
+// This isn't exclusively a problem with invites, though it manifests more clearly there.
+func upClearStuckInvites(ctx context.Context, tx *sql.Tx) error {
+	// The syncv3_unread table is updated any time A) a room is in rooms.join and B) the unread count has changed,
+	// where nil != 0. Therefore, we can use this table as a proxy for "have we seen a v2 response which has put this
+	// room into rooms.join"? For every room in rooms.join, we should have seen a create event for it, and hence have
+	// an entry in syncv3_rooms. If we do not have an entry in syncv3_rooms but do have an entry in syncv3_unread, this
+	// implies we failed to properly store this joined room and therefore the user who the unread marker is for should be
+	// reset to force an initial sync. On matrix.org, of the users using sliding sync, this will catch around ~1.82% of users
+	rows, err := tx.QueryContext(ctx, `
+		SELECT distinct(user_id) FROM syncv3_unread
+		WHERE room_id NOT IN (
+			SELECT room_id
+			FROM syncv3_rooms
+		) GROUP BY user_id
```

Comment:
Is the GROUP BY needed here, given that we're already using DISTINCT?

Reply:
No it isn't.